src/TokenType.h

```diff
@@ -0,0 +1,46 @@
+#ifndef TOKEN_TYPE_H
+#include <string>
+#define TOKEN_TYPE_H
+enum TokenType {
```

Comment:
can we use a `enum class` instead? That we we don't have implicit conversions, and the enum names are local to the enum.
With plain `enum`s:
```
enum Miguel { ff };
enum Jorge {bolinha};

Miguel m = Miguel::ff;
int lol = m;

if (m == Jorge::bolinha) { 
    std::cout << "This will actually be printed\n";
}
```

Weird right 😄 ?

Reply:
Since I'm a noob, i didn't know about `enum classes` 🙃 after reading a little bit about it, for sure makes more sense and I'm already feeling that in the future we would get into trouble due to this . I will change to `enum class`. Thanks!